InnoGotchi.BusinessLogic/Services/CollaboratorService.cs

```diff
@@ -0,0 +1,83 @@
+﻿using AutoMapper;
+using FluentValidation;
+using InnoGotchi.BusinessLogic.Exceptions;
+using InnoGotchi.BusinessLogic.Interfaces;
+using InnoGotchi.Components.DtoModels;
+using InnoGotchi.DataAccess.Interfaces;
+using InnoGotchi.DataAccess.Models;
+using Microsoft.EntityFrameworkCore;
+
+namespace InnoGotchi.BusinessLogic.Services
+{
+    public class CollaboratorService : ICollaboratorService
+    {
+        private readonly IRepository<Collaborator> _collabRep;
+        private readonly IRepository<Farm> _farmRep;
+        private readonly IRepository<IdentityUser> _userRep;
+        private readonly IValidator<CollaboratorDto> _collabValidator;
+        private readonly IMapper _mapper;
+
+        public CollaboratorService(IRepository<Collaborator> collabRep,
+            IRepository<Farm> farmRep,
+            IRepository<IdentityUser> userRep,
+            IValidator<CollaboratorDto> collabValidator,
+            IMapper mapper)
+        {
+            _collabRep = collabRep;
+            _farmRep = farmRep;
+            _userRep = userRep;
+            _collabValidator = collabValidator;
+            _mapper = mapper;
+        }
+
+        public async Task<int?> CreateCollaboratorAsync(CollaboratorDto collaboratorToCreate)
+        {
+            var validationResult = await _collabValidator.ValidateAsync(collaboratorToCreate);
+
+            if (!validationResult.IsValid)
```

Comment:
Add curly braces for if statements everywhere.

Reply:
And you should check all Http methods and replace 200 status code according to their status codes